charts/datadog/templates/agent-services.yaml

```diff
@@ -81,7 +81,7 @@ spec:
       port: {{ .Values.datadog.dogstatsd.port }}
       targetPort: {{ .Values.datadog.dogstatsd.port }}
       name: dogstatsd
-{{- if .Values.datadog.apm.enabled }}
+{{- if or .Values.datadog.apm.enabled .Values.datadog.apm.portEnabled }}
```

Comment:
this if is used in different places, maybe you can make a function to replace it